app/assets/javascripts/components/sampleUpload.jsx

```diff
@@ -267,106 +263,145 @@ class SampleUpload extends React.Component {
   }
 
   isFormInvalid() {
-    if (this.refs.name.value === '' && this.state.selectedProject === 'Select a Project' && this.refs.first_file_source.value === '' && this.refs.second_file_source.value === '' && this.state.selectedHostGenome === '') {
-      this.setState({
-        invalid: true,
-        errorMessage: 'Please fill in all required fields'
-      })
-      return true;
-    } else if (this.refs.name.value === '') {
-        this.setState({
-          invalid: true,
-          errorMessage: 'Please fill in Sample name'
-        })
-        return true;
-    } else if (this.state.selectedProject === 'Select a Project') {
-        this.setState({
-          invalid: true,
-          errorMessage: 'Please select a project'
-        })
-        return true;
-    } else if (this.state.selectedHostGenome === '') {
-      this.setState({
-        invalid: true,
-        errorMessage: 'Please select a host genome'
-      })
-      return true;
+    const errors = {};
+    if(this.state.selectedProject) {
+      if(this.state.selectedProject.toLowerCase() === 'select a project') {
+        errors.selectedProject = 'Please select a project';
+      }
+    } else {
+      errors.selectedProject = 'Please select a project';
```

Comment:
Could you make text like 'select a project' and 'Please select a project' constants so that we only have to change them in the code once if we want to modify them?